### ISSUE TITLE
resample() with how=count causes Segmentation Fault

### ISSUE THREAD

**Issue by kelvin22** [pydata/pandas#10228]
Runing the code as is below results in a segmentation fault when it gets to resample()
The first pass runs, and then either the 2nd or 3rd run produces:

Segmentation fault: 11

It happens on:
- Python 2.7.9 and Pandas 0.16.1
- Python 3.4.3 and Pandas 0.16.1

It does not happen on:
- Python 2.7.9 with Pandas 0.12.0

Those are the only ones I've tested

```
import pandas as pd

frame = pd.DataFrame({'delay_category': {pd.Timestamp('2014-05-18 08:50:00'): 'yes',
  pd.Timestamp('2014-05-19 14:35:00'): 'yes',
  pd.Timestamp('2014-05-22 08:45:00'): 'yes',
  pd.Timestamp('2014-05-27 10:00:00'): 'no',
  pd.Timestamp('2014-05-28 11:05:00'): 'no'}})

frame_filter = frame[frame['delay_category'] == 'da']['delay_category']

for i in range(1,10):
    frame_filter.resample('d', how='count')

```
Setting how=sum also does not result in the issue.

It only happens after the frame has been selected with a parameter that doesn't work, resulting in an empty series.
Eg. this works fine:
```
frame_filter = frame[frame['delay_category'] == 'yes']['delay_category']

for i in range(1,10):
    frame_filter.resample('d', how='count')

```

**Comment by kelvin22** [comment]
Probably related: https://github.com/pydata/pandas/issues/8573

**Comment by jreback** [comment]
yep, looks like a bug. pull-requests are welcome.

**Comment by Garrett-R** [comment]
@kelvin22, great bug report!  Very easy to reproduce.

**Comment by jreback** [comment]
closed by #10337
